modules/hal_nordic/nrfs/dvfs/ld_dvfs_handler.c

```diff
@@ -0,0 +1,378 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "ld_dvfs_handler.h"
+#include "ld_dvfs.h"
+
+#include <hal/nrf_hsfll.h>
+#include <nrfs_dvfs.h>
+#include <nrfs_backend_ipc_service.h>
+
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+LOG_MODULE_DECLARE(LD_DVFS_LIB, CONFIG_LOCAL_DOMAIN_DVFS_LIB_LOG_LEVEL);
+
+K_SEM_DEFINE(dvfs_service_sync_sem, 0, 1);
+K_SEM_DEFINE(dvfs_service_idle_sem, 0, 1);
+
+#define DVFS_SERV_HDL_INIT_DONE_BIT_POS (0)
+#define DVFS_SERV_HDL_FREQ_CHANGE_IN_PROGRESS_BIT_POS (1)
+
+atomic_t dvfs_service_handler_state_bits;
```

Comment:
Is there a reason why these variables are not `static`?

Reply:
It looks like the semaphores above could be made `static`, too.

Reply:
done